clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -4381,6 +3979,101 @@ public boolean hasAllPartitionReportedCompleted() {
     return true;
   }
 
+  public StorageUtilizationManager getStorageUtilizationManager() {
+    return storageUtilizationManager;
+  }
+
+  public AbstractStoreBufferService getStoreBufferService() {
+    return storeBufferService;
+  }
+
+  public SparseConcurrentList<Object> getAvailableSchemaIds() {
+    return availableSchemaIds;
+  }
+
+  public SparseConcurrentList<Object> getDeserializedSchemaIds() {
+    return deserializedSchemaIds;
+  }
+
+  public boolean isGlobalRtDivEnabled() {
+    return isGlobalRtDivEnabled;
+  }
+
+  public Consumer<DataValidationException> getDivErrorMetricCallback() {
+    return divErrorMetricCallback;
+  }
+
+  public abstract Lazy<KeyLevelLocksManager> getKeyLevelLocksManager();
+
+  public abstract KafkaDataIntegrityValidator getKafkaDataIntegrityValidatorForLeaders();
+
+  protected abstract void getAndUpdateLeaderCompletedState(
+      KafkaKey kafkaKey,
+      KafkaMessageEnvelope kafkaValue,
+      ControlMessage controlMessage,
+      PubSubMessageHeaders pubSubMessageHeaders,
+      PartitionConsumptionState partitionConsumptionState);
+
+  protected abstract void updateLatestInMemoryLeaderConsumedRTOffset(
+      PartitionConsumptionState pcs,
+      String ignoredKafkaUrl,
+      long offset);
+
+  protected abstract CompletableFuture<PubSubProduceResult> sendIngestionHeartbeat(
+      PartitionConsumptionState partitionConsumptionState,
+      PubSubTopicPartition topicPartition,
+      PubSubProducerCallback callback,
+      LeaderMetadataWrapper leaderMetadataWrapper,
+      boolean shouldLog,
+      boolean addLeaderCompleteState,
+      LeaderCompleteState leaderCompleteState,
+      long originTimeStampMs);
+
+  protected abstract PubSubMessageProcessedResult processMessage(
+      PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> consumerRecord,
+      PartitionConsumptionState partitionConsumptionState,
+      int partition,
+      String kafkaUrl,
+      int kafkaClusterId,
+      long beforeProcessingRecordTimestampNs,
+      long beforeProcessingBatchRecordsTimestampMs);
+
+  protected PubSubMessageProcessedResult processActiveActiveMessage(
+      PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> consumerRecord,
+      PartitionConsumptionState partitionConsumptionState,
+      int partition,
+      String kafkaUrl,
+      int kafkaClusterId,
+      long beforeProcessingRecordTimestampNs,
+      long beforeProcessingBatchRecordsTimestampMs) {
+    throw new VeniceException("processActiveActiveMessage() should only be called in active active mode");
+  }
```

Comment:
Do you view these functions as needing to be here because it's a work in progress in the migration of the code from `SIT`/`LFSIT` towards `StoreDataPartitionReceiver` (but eventually these functions will also be moved)? Or because this is the meaningful boundary where the handoff between the two classes should take place?